components/CreateGiftCardsForm.js

```diff
@@ -55,9 +56,7 @@ const messages = defineMessages({
 const InlineField = ({ name, children, label, isLabelClickable }) => (
   <Flex flexWrap="wrap" alignItems="center" mb="2.5em" className={`field-${name}`}>
     <Box width={[1, 0.3]}>
-      <label htmlFor={`giftcard-${name}`} style={{ cursor: isLabelClickable ? 'pointer' : 'inherit', width: '100%' }}>
-        {label}
-      </label>
+      <Strong style={{ cursor: isLabelClickable ? 'pointer' : 'inherit', width: '100%' }}>{label}</Strong>
```

Comment:
I don't get this change, `label` is pointing to an `input` (through `htmlFor`). Using `Strong` seems like a regression to me, we lose the link between the label and the input.

Which rule was it breaking?

Reply:
@Betree : The error that we are getting here is, 

```
Form label must have ALL of the following types of associated control: nesting, id  jsx-a11y/label-has-for
```
which points to https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md. 

Now the reason I changed this to `Strong` was that I didn't see anywhere where we use an associated input for this label (correct me if I am wrong please). So I thought that the intended purpose of using a `label` was not quite met in this situation and thus changed it to `Strong`. 😄

Reply:
We should replace https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md by https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md, as the first one seems deprecated. If that creates too many errors we can keep it as a warning for now.

Then about the specific field that this label is targeting, it depends on how `InlineField` is initialized. Here for example: https://github.com/opencollective/opencollective-frontend/blob/8f56712262092a3400f8961a4c8a92a4b81bfa49/components/CreateGiftCardsForm.js#L366

We set the `name` of the `InlineField` to `customMessage`, so `htmlFor` will be equal to `giftcard-customMessage`. If you look a few lines below, you'll see that we initialize the input with `id="giftcard-customMessage"` - properly linking the label to its corresponding input. This is the intended behavior, so we can eventually update the code if the rule does not understand the link but we should not get rid of the `label` as this would lower our accessibility level.

A good test for that (which surfaces that this form is actually missing a few input IDs): click on the label, the input should be selected:

![Peek 2021-03-15 13-15](https://user-images.githubusercontent.com/1556356/111152104-a8e07800-8590-11eb-8662-e8b6086a01f6.gif)

Reply:
@Betree : Thanks for pointing it out. Somehow I've missed the input filed linking and thought those weren't real labels. Agreed and I've corrected it along with many other similar cases. 😯